policyengine_taxsim/config/variable_mappings.yaml

```diff
@@ -442,10 +463,48 @@ policyengine_to_taxsim:
     group_order: 4
     full_text_group: "State Tax Calculation"
     group_column: 1
+    pre_simulation:
+      - ms:
+          implemented: true
+          pre_variable: ms_files_separately
+          variables:
+            - ms_itemized_deductions_indiv
+            - ms_itemized_deductions_joint
+      - ar:
+          implemented: true
+          pre_variable: ar_files_separately
+          variables:
+            - ar_itemized_deductions_indiv
+            - ar_itemized_deductions_joint
+      - mt:
+          implemented: true
+          pre_variable: mt_files_separately
+          variables:
+            - mt_itemized_deductions_indiv
+            - mt_itemized_deductions_joint
+      - ia:
+          implemented: true
+          pre_variable: ia_files_separately
+          variables:
+            - ia_itemized_deductions_indiv
+            - ia_itemized_deductions_joint
+      - ky:
+          implemented: true
+          pre_variable: ky_files_separately
+          variables:
+            - ky_itemized_deductions_indiv
+            - ky_itemized_deductions_joint
     special_cases:
       - il:
           implemented: false
           variable: na_pe
+      - de:
+          implemented: false
+          variable: na_pe
+      - tx:
+          implemented: false
+          variable: na_pe
```

Comment:
```suggestion
```

Texas does not have state income tax

Reply:
right, so the implementation need to avoid extra variable call. This how if this states exists, the code will not execute for those states.

Reply:
Can you elaborate? Do we need to this for every state without income tax?

Reply:
Do we need to keep this?

Reply:
yes, in order to skip executing the variable, as all variable comes from yaml to execute, any cases that needs to skip it required this to ignore.

Reply:
I dont understand, how is Texas (tx) different from other states without state income tax like Florida (fl) or Nevada (nv) or others?

Reply:
If this needs to be done, can you open a separate issue and describe the logic behind it?

Reply:
sounds good to me

Reply:
https://github.com/PolicyEngine/policyengine-taxsim/pull/155 will be working on the generalizing the yaml structure to handle special cases in better way

Reply:
@noman404 can you answer pavel's question?
>I dont understand, how is Texas (tx) different from other states without state income tax like Florida (fl) or Nevada (nv) or others?

Reply:
Yes, we will be having a call for this + #166. 

Yes, we need it for now, how it works currently is, from yaml it essentially mapping against taxsim input keys and calling PE the variable against it, but there are several cases the variable differs state by state and some cases its not available, as like tx, fl, nv do not have income tax. So to avoid extra call and getting exception, we are using special case and pre simulation field to identify them and ignore or call those different variables, as the code does not know which states has which variable or not, all are in yaml config. 

So far I have planned few adjustment that we will discuss and improve the yaml structure. As from our tests for all states, the special cases are quite a long list to maintain [here](https://docs.google.com/spreadsheets/d/1nCGxh3M8xaFnwJRRJKMDTeQomP15f75B/edit?usp=sharing&ouid=107582076828498130762&rtpof=true&sd=true)